src/_pytest/python.py

```diff
@@ -152,7 +152,10 @@ def pytest_configure(config):
 def pytest_pyfunc_call(pyfuncitem):
     testfunction = pyfuncitem.obj
     iscoroutinefunction = getattr(inspect, "iscoroutinefunction", None)
```

Comment:
Unrelated to this change, but this can be made unconditional now that only Python>=3.5 is supported. What you are adding `isasyncgenfunction` is only Python>=3.6, so should stay conditional.